website/docs/cdktf/create-and-deploy/best-practices.mdx

```diff
@@ -55,46 +182,224 @@ You do not always need to create a new class for each stack. For some use cases,
+stage_networking = NetworkingStack(app, "networking-staging", {})
+ecommerce_staging = Ecommerce(app, "ecommerce-staging-us",
+    vpc_id=stage_networking.vpc_id,
+    subnets=stage_networking.subnets,
+    region="us-west-1"
+)
+Ecommerce(app, "ecommerce-staging-eu",
+    vpc_id=stage_networking.vpc_id,
+    subnets=stage_networking.subnets,
+    region="eu-central-1",
+    database_replication_master=ecommerce_staging.database_replication_master
+)
+Blog(app, "ecommerce-staging",
+    vpc_id=stage_networking.vpc_id,
+    subnets=stage_networking.subnets,
+    region="us-west-1"
+)
+app.synth()
+```
+
+```java
+import com.hashicorp.cdktf.App;
+import com.mycompany.app.NetworkingStack;
+import com.mycompany.app.Ecommerce;
+import com.mycompany.app.Blog;
+
+App app = new App();
+NetworkingStack devNetworking = new NetworkingStack(app, "networking-development", Map.of());
+
+// A single region is enough to create a development environment for each product.
+// A single region is enough to create a development environment for each product.
+Ecommerce.Builder.create(app, "ecommerce-development")
+        .vpcId(devNetworking.getVpcId())
+        .subnets(devNetworking.getSubnets())
+        .region("us-west-1")
+        .build();
+Blog.Builder.create(app, "ecommerce-development")
+        .vpcId(devNetworking.getVpcId())
+        .subnets(devNetworking.getSubnets())
+        .region("us-west-1")
+        .build();
+
+// Staging environments require two ecommerce stacks in different regions
+// to test the high availability features of the infrastructure.
+NetworkingStack stageNetworking = new NetworkingStack(app, "networking-staging", Map.of());
+Ecommerce ecommerceStaging = Ecommerce.Builder.create(app, "ecommerce-staging-us")
+        .vpcId(stageNetworking.getVpcId())
+        .subnets(stageNetworking.getSubnets())
+        .region("us-west-1")
+        .build();
+Ecommerce.Builder.create(app, "ecommerce-staging-eu")
+        .vpcId(stageNetworking.getVpcId())
+        .subnets(stageNetworking.getSubnets())
+        .region("eu-central-1")
+        .databaseReplicationMaster(ecommerceStaging.getDatabaseReplicationMaster())
+        .build();
+Blog.Builder.create(app, "ecommerce-staging")
+        .vpcId(stageNetworking.getVpcId())
+        .subnets(stageNetworking.getSubnets())
+        .region("us-west-1")
+        .build();
+app.synth();
+```
+
+```csharp
+using HashiCorp.Cdktf;
+using MyConstructs;
+
+App app = new App();
+NetworkingStack devNetworking = new NetworkingStack(app, "networking-development", new Dictionary<string, object> { });
+
+// A single region is enough to create a development environment for each product.evelopment environment for each product.
+new Ecommerce(app, "ecommerce-development", new EcommerceProps {
+    VpcId = devNetworking.VpcId,
+    Subnets = devNetworking.Subnets,
+    Region = "us-west-1"
+});
+new Blog(app, "ecommerce-development", new BlogProps {
```

Comment:
Those two constructs have the same name which should fail, shouldn't it?
I think another Blog construct further down is also called "ecommerce" in its it as well.

Also true for other languages.